File: problems/algorithms/object-exam-oo/js/response/exam-stats.js
class ExamStats {
  constructor(students) {
    this.students = students;
  }

  avg() {
    let sum = this.students.reduce((sum, student) => sum + student.grade, 0);
    return sum / this.students.length;
  }

  min(count = 1) {
    let grades = this.students.map(student => student.grade);
    grades.sort((a, b) => a - b);
    return grades.slice(0, count);
  }

  max(count = 1) {
    let grades = this.students.map(student => student.grade);
    grades.sort((a, b) => a - b);
    return grades.slice(-count);
  }

  lt(limit) {
    let grades = this.students.map(student => student.grade);
    grades.sort().reverse();
    return grades.filter(grade => grade < limit);
  }

  gt(limit) {
    let grades = this.students.map(student => student.grade);
    grades.sort().reverse();
    return grades.filter(grade => grade > limit);
  }
}

export { ExamStats };

File: problems/algorithms/object-exam-oo/ruby/response/src/exam_stats.rb
class ExamStats
  def initialize(students)
    @students = students
  end

  def avg() 
    sum = 0
    @students.each_with_index do |object|
      sum = sum + object[:grade]
    end
    return sum.to_f / @students.size 
  end

  def min(count = 1) 
    array = @students.map { |n| n[:grade] }
    return array.sort().slice(0,count)
  end

  def max(count = 1) 
    array = @students.map { |n| n[:grade] }
    return array.sort().reverse().slice(0,count)
  end

  def lt(limit) 
    array = @students.map { |n| n[:grade] }
    return array.sort().filter{ |item| item < limit }
  end

  def gt(limit) 
    array = @students.map { |n| n[:grade] }
    return array.sort().filter{ |item| item > limit }
  end
end
